test/test_ion_timestamp.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2009 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "ion_assert.h"
+#include "ion_helpers.h"
+#include "ion_event_util.h"
+#include "ion_test_util.h"
+#include "ion_event_equivalence.h"
+
+struct timestamp_test {
+    const char *str;
+    int local_offset;
+    time_t time;
+    const char *str_unknown_offset;
+};
+
+const int num_tests = 3;
+struct timestamp_test tests[num_tests] = {
```

Comment:
These tests are good candidates for the native parameterization provided by Google test via `TEST_P`. This will eliminate the `for` loops in each test and allow the different cases to be listed and run individually in the IDE. Documentation here: https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#how-to-write-value-parameterized-tests

The simplest example in ion-c is in `test_ion_symbols.cpp`, which contains many [`TEST_P` tests](https://github.com/amzn/ion-c/blob/master/test/test_ion_symbol.cpp#L150) that parameterize on `BinaryAndTextTest`. These parameters are instantiated [here](https://github.com/amzn/ion-c/blob/master/test/test_ion_symbol.cpp#L23), defined in `ion_test_util.h` [here](https://github.com/amzn/ion-c/blob/master/test/ion_test_util.h#L31) and implemented in `ion_test_util.cpp` [here](https://github.com/amzn/ion-c/blob/master/test/ion_test_util.cpp#L20).

Reply:
I haven't been able to parameterize these tests using a struct due to a compilation error I'm not smart enough to fix.  Adding logic to provide names for each parameterized iteration of a test prevents the tests from showing in the CLion test display, so I've removed that aspect from the latest commit.

I appreciate parameterized tests, but I'm not fully sold on applying the idea here.  It mostly appears to add complexity for what is [currently] a fairly simple set of tests.  As this set of tests grows in size/complexity, perhaps parameterization would add more value.  What do you think?

Reply:
I still prefer the parameterized versions. I don't think there's a reason to undo that work now.

I encountered the problem of CLion not playing nicely with the named test instances in test_vectors, too. There, I hid the verbose names behind an ifdef that is not defined when running in CLion, allowing for the names to be printed when built elsewhere (the command line, Travis/AppVeyor). I don't think that's a big enough value add to spend time on here, though; the biggest benefit of parameterized tests is being able to run the particular instance that failed individually, which is still possible even without nice names.

Reply:
Ok, still parameterized but without names, and the build now succeeds.